src/base/tests/test_util.py

```diff
@@ -1457,6 +1468,23 @@ def test_SelfPrint_prepare(self, value):
         evaluated = safe_eval(replaced_value, ctx, nocopy=True)
         self.assertEqual(str(evaluated), value)
 
+    @parametrize(
+        [
+            (value,)
+            for value in [
+                # iterators
+                "[a.b for a in b]",
+                "4 in b",
+                # bool conversions
+                "not a",
```

Comment:
For the record, `and`/`or` was also wrong since ever :/
```suggestion
                "not a",
                "a and b",
                "a or b"
```